app/test/consistent_apphash_test.go

```diff
@@ -0,0 +1,173 @@
+
+	// Get the name of the records
+	for _, rec := range recs {
+		accountNames = append(accountNames, rec.Name)
+	}
+
+	// Apply genesis state to the app.
+	_, _, err = testutil.ApplyGenesisState(testApp, pubKeys, 1_000_000_000, app.DefaultInitialConsensusParams())
+	require.NoError(t, err)
+
+	// Query keyring account infos
+	accountInfos := queryAccountInfo(testApp, accountNames, kr)
+
+	// Create accounts for the signer
+	accounts := make([]*user.Account, 0, len(accountInfos))
+	for i, accountInfo := range accountInfos {
+		account := user.NewAccount(accountNames[i], accountInfo.AccountNum, accountInfo.Sequence)
+		accounts = append(accounts, account)
+	}
+
+	// Create a signer with keyring accounts
+	signer, err := user.NewSigner(kr, enc.TxConfig, testutil.ChainID, app.DefaultInitialVersion, accounts...)
+	require.NoError(t, err)
+
+	amount := sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewIntFromUint64(1000)))
+
+	// Create an SDK Tx
+	sdkTx := sdkTxStruct{
+		sdkMsgs: []sdk.Msg{
+			banktypes.NewMsgSend(signer.Account(accountNames[0]).Address(),
+				signer.Account(accountNames[1]).Address(),
+				amount),
+		},
+		txOptions: blobfactory.DefaultTxOpts(),
+	}
+
+	// Create a Blob Tx
+	blobTx := blobTxStruct{
+		author:    accountNames[2],
+		blobs:     []*blob.Blob{blob.New(Namespace(), []byte{1}, appconsts.DefaultShareVersion)},
+		txOptions: blobfactory.DefaultTxOpts(),
+	}
+
+	// Create SDK Tx
+	rawSdkTx, err := signer.CreateTx(sdkTx.sdkMsgs, sdkTx.txOptions...)
+	require.NoError(t, err)
+
+	// Create Blob Tx
+	rawBlobTx, _, err := signer.CreatePayForBlobs(blobTx.author, blobTx.blobs, blobTx.txOptions...)
+	require.NoError(t, err)
+
+	// BeginBlock
+	header := tmproto.Header{
+		Version: version.Consensus{App: 1},
+		Height:  testApp.LastBlockHeight() + 1,
+	}
+	testApp.BeginBlock(abci.RequestBeginBlock{Header: header})
+
+	// Deliver SDK Tx
+	resp := testApp.DeliverTx(abci.RequestDeliverTx{Tx: rawSdkTx})
+	require.EqualValues(t, 0, resp.Code, resp.Log)
+
+	// Deliver Blob Tx
+	blob, isBlobTx := blob.UnmarshalBlobTx(rawBlobTx)
+	require.True(t, isBlobTx)
+	resp = testApp.DeliverTx(abci.RequestDeliverTx{Tx: blob.Tx})
+	require.EqualValues(t, 0, resp.Code, resp.Log)
+
+	// EndBlock
+	testApp.EndBlock(abci.RequestEndBlock{Height: header.Height})
+
+	// Commit the state
+	testApp.Commit()
+
+	// Get the app hash
+	appHash := testApp.LastCommitID().Hash
+
+	// Require that the app hash is equal to the app hash produced on a different commit
+	require.Equal(t, expectedAppHash, appHash)
+}
+
+// DeterministicNamespace returns a deterministic namespace
+func Namespace() appns.Namespace {
+	return appns.Namespace{
+		Version: 0,
+		ID:      []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 37, 67, 154, 200, 228, 130, 74, 147, 162, 11},
+	}
+}
+
+// DeterministicKeyRing returns a deterministic keyring and a list of deterministic public keys
+func DeterministicKeyRing(cdc codec.Codec) (keyring.Keyring, []types.PubKey) {
```

Comment:
[optional] un-export

```suggestion
// deterministicKeyRing returns a deterministic keyring and a list of deterministic public keys
func deterministicKeyRing(cdc codec.Codec) (keyring.Keyring, []types.PubKey) {
```